src/components/SuperDrawer/useSuperDrawer.tsx

```diff
@@ -64,40 +64,48 @@ export function useSuperDrawer(): UseSuperDrawerHook {
   } = useContext(BeamContext);
   const { openModal } = useModal();
 
+  /**
+   * Validates whether a new Superdrawer can be opened / whether the current Superdrawer can be closed.
+   * Breaking the validation into a separate function allows openInDrawer
+   * to validate the drawer's state without closing it
+   * */
+  function validateDrawerChange (onClose: () => void) {
```

Comment:
Maybe we could make this:

```
function validateDrawerChange(doChange: () => void): void {
  ...
```

Note that besides the `onClose` --> `doChange` rename (open to better names for `doChange`), specifically using a `void` return type...

I know you're just moving this code around, but the previous "sometimes return true, sometimes invoke confirm, sometimes return false" behavior was kind of confusing to me...

I think a potentially better api is to instead of sometimes checking the return value/sometimes not, just assume that `doChange` is goingn to be called...

We might have to change the `useSuperDrawer.closeDrawer` from boolean to void, but I think that'd be a good change to make anyway...

Reply:
I like the change! I'll make the update.

Reply:
+1! Wondering if we can change the name to something that can match validating if we can close the **drawer** and if we can close the **detail drawer**.

I am biased to `canCloseDrawer(() => {...})` and `canCloseDrawerDetail(i, () => {...})`

Reply:
@KoltonG if you're thinking of names for `validateDrawerChange` I was thinking of `maybeChangeDrawer`... or `changeDrawerIfConfirmed`...

Reply:
@KoltonG  are you referring to pulling out the `for` loops that check the drawerDetails, or the whole validation function. 
@stephenh I'm cool with `maybeChangeDrawer`

Reply:
This was for breaking out the check code for:
- Validating if a drawer detail can be closed
- Validating if a drawer can be closed (which also includes validating that each drawer detail can be closed -> which could use the first validator).

So closeDrawerDetails would be something like:
```
validateDrawerDetailChange(contentStack.current.length - 2, () => {... code to close detail drawer and modal})
```

And the current `validateDrawerChange` would include `validateDrawerDetailsChange` in it's loop to check if each drawer details can be closed!

Reply:
@KoltonG since the `canCloseChecks` are only being iterated over in one place, with the exception of readability, I'm not sure I see a ton of benefit in further breaking things out. Would you want to talk over it at some point today?